store/test-store/devel/up.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+
+set -e
+
+ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+keep=
+
+main() {
+  pushd "$ROOT" &> /dev/null
+
+  while getopts "hk" opt; do
+    case $opt in
+      h) usage && exit 0;;
+      k) keep=true;;
+      \?) usage_error "Invalid option: -$OPTARG";;
+    esac
+  done
+  shift $((OPTIND-1))
+
+  if [[ -d "./data" && $keep == "" ]]; then
+    echo "Cleaning data directory"
+    rm -rf ./data 1> /dev/null
+  fi
+
+  prepare
+
+  # Pass execution to docker compose
+  exec docker-compose up
+}
+
+prepare() {
+  if [[ ! -d "./data/ipfs" ]]; then
+    mkdir -p ./data/ipfs 1> /dev/null
+  fi
+
+  if [[ ! -d "./data/postgres" ]]; then
+    mkdir -p ./data/postgres 1> /dev/null
+  fi
+}
+
+usage_error() {
+  message="$1"
+  exit_code="$2"
+
+  echo "ERROR: $message"
+  echo ""
+  usage
+  exit ${exit_code:-1}
+}
+
+usage() {
+  echo "usage: up [-c]"
+  echo ""
+  echo "Setup required files layout and launch 'docker compose up'"
+  echo "spinning up all required development dependencies."
+  echo ""
+  echo "Options"
+  echo "    -k          Clean 'data' directory before launching dependencies"
```

Comment:
It was indeed like that at the start but I reverted the flag since it was made for integration tests, which I think requires to start fresh each time.

If it's indeed required to start fresh on each run, I think keeping the current behavior (but fixing the help message) is good. Otherwise, I fully agree that the older `-c` option is a better default.

Tell me what you think.

Reply:
Can't the integration tests just pass `-c` when they invoke this script? I am actually not sure which integration tests you are referring to.

Reply:
What I meant is, since the script is expected to be run in such sequence:

```
./up.sh
GRAPH_NODE_TEST_CONFIG=... cargo test # Runs integration tests
```

It felt preferable that the `up.sh` script always starts "fresh". Currently, there is no automatic invocation of `up.sh` anywhere, it's a manual process.

I thought doing:

```
./up.sh
GRAPH_NODE_TEST_CONFIG=... cargo test # Runs integration tests
GRAPH_NODE_TEST_CONFIG=... cargo test # Runs integration tests
```

Would bark on the second invocation of `cargo test`. I should have tested that :)

Reply:
The tests are very careful to make sure that they set up a known good state regardless of what state the database was in before. In my workflow, cleaning out the database for the tests is a relatively rare occurrence.

Reply:
Ok perfect, that's what I wanted to understand.

I'll revert to using `-c` and clean only on demand.